providers.d/bandersnatch/provider_wrapper.sh

```diff
@@ -12,16 +12,35 @@ bashelliteProviderWrapperBandersnatch() {
 
   local directory_parameter="$(grep -oP "(?<=(^directory = )).*" ${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf)"
   if [[  "${directory_parameter}" != "${_r_mirror_tld}/${_n_repo_name}" ]]; then
-    utilMsg FAIL "$(utilTime)" "The \"directory\" parameter (${directory_parameter}) in provider.conf does not match mirror location (${_r_mirror_tld}/${_n_repo_name}); exiting."
+    utilMsg WARN "$(utilTime)" "The \"directory\" parameter (${directory_parameter}) in provider.conf does not match mirror location (${_r_mirror_tld}/${_n_repo_name})..."
   fi
 
+  local config_file="${_r_metadata_tld}/repos.conf.d/${_n_repo_name}/provider.conf"
+  local tmp_config_file="${HOME}/.bashellite/tmp_${_n_repo_name}_provider.conf"
+
+  > ${tmp_config_file}
+
+  local line_counter=0
+  while IFS=$'\n' read line; do
+    if [[ ${line} =~ ^directory[[:blank:]]*[=][[:blank:]]*[[:alnum:]/_-.]+ ]]; then
+      if [ "${line_counter}" = 0 ]; then
+        echo "directory = ${_r_mirror_tld}/${_n_repo_name}" >> ${tmp_config_file}
+      else
+        utilMsg WARN "$(utilTime)" "Duplicate directory parameter found.  Skipping line..."
+      fi
+    else
+      echo "${line}" >> ${tmp_config_file}
+    fi
+    line_counter=$(($line_counter + 1))
```

Comment:
This line needs to be inside the while loop near your `echo "directory` line. Right now, if the first line in the config file is something besides `^directory =`, your line_counter variable will get incremented, and your line_counter test will always hit your `else` conditional for each subsequent line.